types/hooks/HookContext.d.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * Base class for all hook context objects
+ */
+export declare class HookContext {
+    /**
+     * For testing purposes only. This will always be constructed for you when run in the context of the Azure Functions runtime
+     */
+    constructor(init?: HookContextInit);
+
+    /**
+     * The recommended place to store and share data between hooks in the same scope (app-level vs invocation-level).
+     * You should use a relatively unique property name so that it doesn't conflict with other hooks' data.
```

Comment:
Nit: I think it's better to drop `relatively`. The more unique, the safer, right?